### PR TITLE
hotfix(state-access): Guards was added

### PR DESCRIPTION
**Pull request by moiskillnadne**
### 📝 Description

🔗 [Jira Ticket M2-6534](https://mindlogger.atlassian.net/browse/M2-6534)

One of the new features includes state changes but at some specific moments, the application had the old version of the state. Guard was added to validate that the required state exists.

Changes include:

- Multiinformant feature

### CODE REVIEW

File: src/features/TakeNow/ui/ValidateTakeNowParams.tsx [hunk 1]
```diff
@@ -31,13 +31,13 @@ function ValidateTakeNowParams({
...
-        sourceSubject.id !== multiInformantState.sourceSubject?.id ||
-        targetSubject.id !== multiInformantState.targetSubject?.id
+        sourceSubject.id !== multiInformantState?.sourceSubject?.id ||
+        targetSubject.id !== multiInformantState?.targetSubject?.id
       ) {
         initiateTakeNow({ sourceSubject, targetSubject });
       }
     }
-  }, [data, initiateTakeNow, isInMultiInformantFlow, isSuccess]);
+  }, [data, getMultiInformantState, initiateTakeNow, isInMultiInformantFlow, isSuccess]);
```
Comment (sultanofcardio) [inline src/features/TakeNow/ui/ValidateTakeNowParams.tsx:40]: I believe we should leave the dependency arrays as-is. Since these changes are unrelated to the bug fix, they can be added to dev instead

File: src/widgets/ActivityDetails/model/hooks/useAnswers.ts [hunk 2]
```diff
@@ -155,18 +155,20 @@ export const useAnswer = (props: Props) => {
...
+      props.activityId,
       props.eventId,
       props.eventsRawData,
-      props.flowId,
+      encryptPayload,
+      getGroupProgress,
+      featureFlags.enableMultiInformant,
+      generateUserPrivateKey,
       getMultiInformantState,
+      isInMultiInformantFlow,
```
Comment (sultanofcardio) [inline src/widgets/ActivityDetails/model/hooks/useAnswers.ts:171]: Same here. Can we revert the changes to the dependency array?

File: src/features/TakeNow/ui/ValidateTakeNowParams.tsx (hunk 1 above)
Comment (moiskillnadne) [reply src/features/TakeNow/ui/ValidateTakeNowParams.tsx]: Thank you, makes sense. Done

File: src/widgets/ActivityDetails/model/hooks/useAnswers.ts (hunk 2 above)
Comment (moiskillnadne) [reply src/widgets/ActivityDetails/model/hooks/useAnswers.ts]: Done

**Review by sultanofcardio — APPROVED** [review]
